### PR TITLE
creating entry for turbo codes; also working out how to create a pull request

### PR DESCRIPTION
**Pull request by jonlin1000**

### CODE REVIEW

File: codes/classical/bits/turbo.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+name: 'Turbo code'
+introduced: '\cite{doi:10.1109/ICC.1993.397441}'
+
+description: 'Codes with composite structure, where information bits are encoded with many different component codes. Between encodings, these codes are permuted in an interleaving fashion.'
+
+protection: 'Parallel concatenated turbo codes have typical minimum distance with upper bound \(O(\log(n))\). Truhachev, Lentmacher, and Zignagirov produced a sequence of turbo codes with minimum distance \(\Theta(\log(n))\). \cite{doi:10.1109/ISIT.2002.1023356}'
+
+features:
+
+  rate: 'The natural weight of a turbo code is \(R = 1/3\). The rate of transmission can be changed depending on the method of puncturing removal \cite{doi:10.12928/telkomnika.v10i4.171}.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/turbo.yml:20]: The natural **rate**?

File: codes/classical/bits/turbo.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  encoders:
+    - 'Many choices can be made in encoding, including how to achieve termination and whether tail bits are transmitted or not, and choice of permutation.'
+
+  #  decoders:
+  #    - ''
+
+  threshold:
+    - 'Berrou et al. achieved performance on turbo codes which was \(0.7dB\) from the Shannon channel capacity limit. \cite{doi:10.1109/ICC.1993.397441}'
```
Comment (valbert4) [inline codes/classical/bits/turbo.yml:29]: This goes into rate.

File: codes/classical/bits/turbo.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#this entry is far from complete. expect updates in the following days.
+
+code_id: turbo
+physical: bits
+logical: bits
+
+name: 'Turbo code'
+introduced: '\cite{doi:10.1109/ICC.1993.397441}'
+
+description: 'Codes with composite structure, where information bits are encoded with many different component codes. Between encodings, these codes are permuted in an interleaving fashion.'
```
Comment (valbert4) [inline codes/classical/bits/turbo.yml:14]: This is not precise enough to give me a feel for these codes. Please try a new first sentence + also a more technical defn. See 21.8.1 Huffman Kim Sole.

File: codes/classical/bits/turbo.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+realizations:
+  - 'Turbo codes are recommended in Consultative Committee on Space Data Systems (CCDJS)\cite{doi:10.2514/6.2002-T5-11}.'
+  - 'Turbo codes are used in several standards related to wireless communication, including W-CDMA, DVB-RCS, TD-SCDMA, 802.16, and cdma2000 \cite{doi:10.1109/JPROC.2007.895197}.'
+
+#notes:
+
+relations:
+  parents:
+    - code_id: convolutional
```
Comment (valbert4) [inline codes/classical/bits/turbo.yml:40]: inner + outer codes should be cousins. parent is parallel_concatenated + smallest code family that contains this one (binary_linear?)

**Pull request closed without merging**